lib/MusicBrainz/Server/ControllerBase/WS/2.pm

```diff
@@ -103,7 +103,12 @@ sub method_not_allowed : Private {
     ));
 }
 
-sub begin : Private { }
+sub begin : Private {
+    my ($self, $c) = @_;
+
+    $c->stash->{serializer} = $self->get_serialization($c);
```

Comment:
This previously happened in `validate`, but that's only run if we already have a valid path. We need a serializer even for unknown paths, so it has to be initialized in `begin` now.